app/src/main/java/io/intrepid/contest/screens/contestcreation/namecontest/NameContestPresenter.java

```diff
@@ -13,15 +14,34 @@
     }
 
     @Override
+    public void onViewCreated() {
+        super.onViewCreated();
+            view.disableNext();
+    }
+
+    @Override
     public void onContestNameUpdate(String contestName) {
-        if(contestName.isEmpty()){
-           showError();
-        }else{
-          view.saveEnteredName(contestName);
+        view.saveEnteredName(contestName);
+    }
+
+    @Override
+    public void onNextInvalidated() {
+        view.disableNext();
+    }
+
+    @Override
+    public void onNextValidated() {
+        if (view != null) {
+            view.enableNext();
         }
     }
 
-    private void showError() {
-        view.showError();
+    @Override
+    public void onTextChanged(CharSequence newName) {
+        if (TextUtils.isEmpty(newName)) {
```

Comment:
I'm already checking this in my Presenter or did you mean check in the NewContestPresenter rather than the NameContestPresenter. The reason this work is being done within this presenter is that my NameContestPresenter is directly tied to the view for naming a category, and that screen has to hide or show the next button depending on whats already been typed in. this is more for checking the text as it is altered at any time rather than the final submission. Does that make sense, @jmccormack200